internal/engine/wazevo/e2e_test.go

```diff
@@ -122,6 +122,128 @@ func TestE2E(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "divrem_unsigned_return", m: testcases.DivUReturn32.Module,
+			calls: []callCase{
+				{
+					params:     []uint64{21, 10, 21, 10},
+					expResults: []uint64{21 / 10, 21 % 10},
+				},
+				{
+					params: []uint64{21, 0, 1, 1},
+					expErr: "wasm error: integer divide by zero",
+				},
+				{
+					params: []uint64{1, 1, 21, 0},
+					expErr: "wasm error: integer divide by zero",
+				},
+				{
+					params: []uint64{
+						0x80000000, 0xffffffff, 1, 1,
+					},
+					expResults: []uint64{0, 0},
+				},
+				{
+					params: []uint64{
+						1, 1, 0x80000000, 0xffffffff,
+					},
+					expResults: []uint64{1, 0x80000000},
+				},
+			},
+		},
+		{
+			name: "divrem_unsigned_return", m: testcases.DivUReturn64.Module,
+			calls: []callCase{
+				{
+					params:     []uint64{21, 10, 21, 10},
+					expResults: []uint64{21 / 10, 21 % 10},
+				},
+				{
+					params: []uint64{21, 0, 1, 1},
+					expErr: "wasm error: integer divide by zero",
+				},
+				{
+					params: []uint64{1, 1, 21, 0},
+					expErr: "wasm error: integer divide by zero",
+				},
+				{
+					params:     []uint64{0x80000000, 0xffffffff, 1, 1},
+					expResults: []uint64{0, 0},
+				},
+				{
+					params:     []uint64{1, 1, 0x80000000, 0xffffffff},
+					expResults: []uint64{1, 0x80000000},
+				},
+			},
+		},
+		{
+			name: "divrem_signed_return32", m: testcases.DivSReturn32.Module,
+			calls: []callCase{
+				{
+					params:     []uint64{21, 10, 21, 10},
+					expResults: []uint64{21 / 10, 21 % 10},
+				},
+				{
+					params: []uint64{21, 0, 1, 1},
+					expErr: "wasm error: integer divide by zero",
+				},
+				//{ // This fails.
+				//	params: []uint64{1, 1, 21, 0},
+				//	expErr: "wasm error: integer divide by zero",
+				//},
+				{
+					params: []uint64{0x80000000, 0xffffffff, 1, 1},
+					expErr: "wasm error: integer overflow",
+				},
+				{
+					params:     []uint64{1, 1, 0x80000000, 0xffffffff},
+					expResults: []uint64{1, 0},
+				},
+			},
+		},
+
+		{
+			name: "divrem_signed_return32 inverted rem div order (different crash)", m: testcases.DivSReturn32_weird.Module,
+			calls: []callCase{
+				{
+					params: []uint64{21, 0, 1, 1},
+					expErr: "wasm error: integer divide by zero",
+				},
+				{ // This now succeeds.
+					params: []uint64{1, 1, 21, 0},
+					expErr: "wasm error: integer divide by zero",
+				},
+				//{ // This now fails.
+				//	params: []uint64{0x80000000, 0xffffffff, 1, 1},
+				//	expErr: "wasm error: integer overflow",
+				//},
+				{
```

Comment:
it seems like if I have more than one error trap in my function, and the generated code is long enough than the error handers down low are generated incorrectly and they SIGSEGV. In fact, this same e2e test above has DivS,RemS and (rem 21%0) crashed. This test case here (Rems, DivS) does not crash and raises the error correctly, but fails on integer overflow (which was handled correctly in the example above)